logviewer/src/main/java/com/nodesagency/logviewer/data/database/DatabaseLogRepository.kt

```diff
@@ -74,4 +79,22 @@ internal class DatabaseLogRepository(
             .logDetailsDao()
             .getLogDetails(logEntryId)
     }
+
+
+    override fun getLogsFilteredBy(state: FilterState): DataSource.Factory<Int, LogEntry> {
```

Comment:
It would be nice to cover this and `getCategoriesByName` with tests if you have some time. Since they are a bit smarter than the rest of the class, it would be good to assure they're performing 100% correctly. You can use [LoggerTest.kt](https://github.com/nodes-android/lib-log-viewer/blob/develop/logviewer/src/androidTest/java/com/nodesagency/logviewer/LoggerTest.kt) for reference. You can also reuse its code for in-memory DB creation etc.

The other methods are more trivial so I was lazy to add tests for those. Shame on me and I should have done it regardless, will make up for it at some point :)

Reply:
Agreed :D